recipes/newrelic/apm/php-agent/php.yml

```diff
@@ -67,3 +70,47 @@ install:
           else
             echo "You will need to restart your PHP application in order for instrumentation to be enabled."
           fi
+
+    send_transaction:
+      cmds:
+        - echo "Send queryable transactions."
+        - |
+          TMP_INSTALL_DIR="/tmp/php-agent-install"
```

Comment:
```suggestion
          TMP_INSTALL_DIR=$(mktemp -d -t php-agent-install-XXXXX)
```

Reply:
This is assuming the tmp install directory was already created and used in prior steps (it assumes the install ran install from that directory).  Should not be created here, but ideally created in prior step and referenced here.

Reply:
Using a randomized temp directory name helps guard against multiple installs conflicting with one another and makes certain attacks slightly more difficult to execute.

In order for us to be able to do this, we will need to be able to either access env vars set by previous steps or will need to be able to share shell vars across steps. Is a fresh shell env used for each step or do they all share a common one?

Reply:
Good point about the possibility for attack against known directory names.

I wasn't sure about fresh shell vs shared, but since I've seen other agents use text to share info so I'd assumed fresh shell.

Although the directory will be created/destroyed each run, because of how the agent would be downloaded/installed, multiple install attempts could theoretically exist in the same dir (i.e., the log file from installation attempts is timestamped) and why the script looks for the most recent log file not just a log file.  

I think it is a fresh shell so it'd be hard to pass the var name between tasks.  Also, given it's created/torn down with each run, slightly less worried about probability of attack.

Reply:
Added blocks for now that will create/delete similar to how dotnet does [creates](https://github.com/newrelic/open-install-library/blob/e7347282f607e079f528d8199b678f29dbeb4fcb/recipes/newrelic/apm/dotNet/linux-systemd.yml#L129) / [deletes](https://github.com/newrelic/open-install-library/blob/e7347282f607e079f528d8199b678f29dbeb4fcb/recipes/newrelic/apm/dotNet/linux-systemd.yml#L336).  Exact naming scheme can still be under discussion.

Reply:
Looks like there's a way to declare variables in recipes. Here's an example from a Java recipe Gabe found: https://github.com/newrelic/open-install-library/blob/0f621414cda830c4eea15752e4ad65847df82897/recipes/newrelic/apm/java/linux.yml#L32

Reply:
Cool, thanks!  I'll get that integrated.